### PR TITLE
Final Voting migration + output html

### PR DESCRIPTION
**Pull request by Raemon**
EDIT: This PR has been substantially updated. It now also serves the purpose of generating the html for the final list of all posts, plus their donation buttons.

### CODE REVIEW

File: packages/lesswrong/lib/collections/posts/schema.ts [hunk 1]
```diff
@@ -669,6 +670,58 @@ const schema: SchemaType<DbPost> = {
     optional: true,
   },
 
+
+  reviewVoteScoreHighKarma2: {
```
Comment (jpaddison3) [inline packages/lesswrong/lib/collections/posts/schema.ts:674]: Right about here I'd put a comment explaining what these are for.

File: packages/lesswrong/server/migrations/2022-01-12-updateQuadraticVotes-2.ts [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    cost: 45,
+    value: 9 
+  },
+}
+
+const getCost = (vote) => voteMap[vote.qualitativeScore].cost
+const getValue = (vote) => voteMap[vote.qualitativeScore].value
+
+// TODO: Write a better version of this migration which properly normalizes vote strength
+registerMigration({
```
Comment (jpaddison3) [inline packages/lesswrong/server/migrations/2022-01-12-updateQuadraticVotes-2.ts:44]: I assume this is entirely copy-pasta-ed without change and I don't need to review it?

Comment (jpaddison3) [reply packages/lesswrong/server/migrations/2022-01-12-updateQuadraticVotes-2.ts:44]: (Copy pasta is fine here, IMO, as migrations are one and done in this codebase.)

**Comment by jpaddison3** [comment]
I'm sorry, I'm not sure what these are for. Why do we need to differentiate from the originals?

File: packages/lesswrong/lib/collections/posts/fragments.ts [hunk 3]
```diff
@@ -97,25 +97,35 @@ registerFragment(`
...
     positiveReviewVoteCount
+
     reviewVoteScoreAllKarma
     reviewVotesAllKarma
     reviewVoteScoreHighKarma
     reviewVotesHighKarma
     reviewVoteScoreAF
     reviewVotesAF
 
+    reviewVoteScoreHighKarma2
```
Comment (darkruby501) [inline packages/lesswrong/lib/collections/posts/fragments.ts:111]: What, what's up with the 2? What does the 2 mean?

Comment (darkruby501) [reply packages/lesswrong/lib/collections/posts/fragments.ts:111]: Okay, I see. I'd rename from "2" to "2ndRound"

**Review by darkruby501 — CHANGES_REQUESTED** [review]
I haven't reviewed the migration since I don't know what it's about.

WIth regard to the new fields, they need a clearer name than "2".

File: packages/lesswrong/server/migrations/2022-01-30-updateFinal2020ReviewVotes.ts [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+
+    const finalPosts = await Posts.find({reviewCount: {$gt: 0}, finalReviewVoteScoreHighKarma: {$exists: true}, postedAt: {$gte: moment(`${REVIEW_YEAR}-01-01`).toDate()}}, {sort: {finalReviewVoteScoreHighKarma: -1}}).fetch()
+
+    const authorIds = finalPosts.map(post => post.userId)
+
+    const authors = await Users.find({_id: {$in:authorIds}}).fetch() 
+
+    const getAuthor = (post) => authors.filter(author => author._id === post.userId)[0]
+
+    const primaryColor = "#5f9b65" // ea-forum-look-here
```
Comment (jpaddison3) [inline packages/lesswrong/server/migrations/2022-01-30-updateFinal2020ReviewVotes.ts:97]: I get this when I go to run it and submit a separate PR. Thanks for calling it out.

**Review by jpaddison3 — APPROVED** [review]
I haven't tested this locally, but I believe if it runs and produces something that looks good, then it's probably good.

File: packages/lesswrong/server/migrations/2022-01-12-updateQuadraticVotes-2.ts [hunk 5]
```diff
@@ -0,0 +1,109 @@
+import { registerMigration } from './migrationUtils';
```
[the remark this answers is not in the document]
Comment (Raemon) [reply packages/lesswrong/server/migrations/2022-01-12-updateQuadraticVotes-2.ts:1]: Ah yeah. Thanks.

**Comment by Raemon** [comment]
@jpaddison3 btw you requested changes so you'll need to approve

**Review by darkruby501 — APPROVED** [review]
I didn't read this in detail but nothing objectionable stuck out to me.

**Pull request merged**